File: util/src/main/java/com/zaze/utils/DescriptionUtil.java
package com.zaze.utils;

/**
 * Description : 用于转换生成一些描述的工具类
 *
 * @author : ZAZE
 * @version : 2017-12-28 - 10:32
 */
public class DescriptionUtil {

    /**
     * 转换成字节单位
     *
     * @param byteLength 字节长度
     * @return 格式化后输出
     */
    public static String toByteUnit(long byteLength) {
        long kb = byteLength >> 10;
        long mb = byteLength >> 20;
        long gb = byteLength >> 30;
        if (gb > 0) {
            return ZStringUtil.format("%.3fGB", mb * 1.0f / 1024);
        } else if (mb > 0) {
            return ZStringUtil.format("%.3fMB", kb * 1.0f / 1024);
        } else if (kb > 0) {
            return ZStringUtil.format("%dKB", kb);
        }
        return ZStringUtil.format("%dB", byteLength);
    }
}

File: util/src/main/java/com/zaze/utils/DescriptionUtil.kt
package com.zaze.utils

import com.zaze.utils.ZStringUtil

/**
 * Description : 用于转换生成一些描述的工具类
 *
 * @author : ZAZE
 * @version : 2017-12-28 - 10:32
 */
object DescriptionUtil {
    /**
     * 转换成字节单位
     *
     * @param byteLength 字节长度
     * @return 格式化后输出
     */
    @JvmStatic
    fun toByteUnit(byteLength: Long, unit: Int = 1024): String {
        val kb = byteLength / unit
        val mb = byteLength / unit / unit
        val gb = byteLength / unit / unit / unit
//        val kb = byteLength shr 10
//        val mb = byteLength shr 20
//        val gb = byteLength shr 30
        return when {
            gb > 0 -> {
                ZStringUtil.format("%.2fGB", mb * 1.0f / unit)
            }
            mb > 0 -> {
                ZStringUtil.format("%.2fMB", kb * 1.0f / unit)
            }
            kb > 0 -> {
                ZStringUtil.format("%dKB", kb)
            }
            else -> ZStringUtil.format("%dB", byteLength)
        }
    }
}